### PR TITLE
switch to use mutex in AbstractHandle.wait method

### PR DESCRIPTION
**Pull request by nijikon**
Closes https://github.com/karafka/rdkafka-ruby/issues/390

### CODE REVIEW

File: lib/rdkafka/abstract_handle.rb [hunk 1]
```diff
@@ -45,7 +54,7 @@ def pending?
     # on the operation. In this case it is possible to call wait again.
     #
     # @param max_wait_timeout [Numeric, nil] Amount of time to wait before timing out.
-    #   If this is nil it does not time out.
+    #   If this is nil we will use WAIT_TIMEOUT value
```
Comment (mensfeld) [inline lib/rdkafka/abstract_handle.rb:57]: "If this is nil, will wait forever" would be more explicit IMHO. WAIT_TIMEOUT is an implementation detail

File: lib/rdkafka/callbacks.rb [hunk 2]
```diff
@@ -157,6 +157,8 @@ def self.process_create_topic(event_ptr)
           create_topic_handle[:error_string] = create_topic_results[0].error_string
           create_topic_handle[:result_name] = create_topic_results[0].result_name
           create_topic_handle[:pending] = false
+
+          create_topic_handle.broadcast
```
Comment (mensfeld) [inline lib/rdkafka/callbacks.rb:161]: can we name it "unlock" or something similar? broadcast seems outside of our domain

**Review by mensfeld — CHANGES_REQUESTED** [review]

**Review by mensfeld — APPROVED** [review]

**Comment by nijikon** [comment]
@thijsc will you have time this coming week to look at this change?

**Review by thijsc — APPROVED** [review]
This looks good to me.

**Pull request merged**